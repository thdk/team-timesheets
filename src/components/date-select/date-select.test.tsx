import React from "react";
import { render } from "@testing-library/react";
import { DateSelect } from "./";

beforeAll(() => {
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date(2020, 3, 1));
});

afterAll(() => {
    jest.useRealTimers();
});

describe("DateSelect", () => {
    it("should render", () => {
        const onMonthChange = jest.fn();
        const onYearChagne = jest.fn();

        const { asFragment } = render(
            <DateSelect
                onMonthChange={onMonthChange}
                onYearChange={onYearChagne}
            />
        );

        expect(asFragment()).toMatchSnapshot();

    });

    it("should render with given year and month selected", () => {
        const yearValue = 2019;
        const monthValue = 3;

        const { container } = render(
            <DateSelect
                onMonthChange={jest.fn()}
                onYearChange={jest.fn()}
                month={monthValue}
                year={yearValue}
            />
        );

        const yearEl = container.querySelector(".date-select-year .mdc-select__selected-text");
        expect(yearEl).not.toBeNull();

        expect(yearEl!.innerHTML).toBe(yearValue.toString());

        const monthEl = container.querySelector(".date-select-month .mdc-select__selected-text");
        expect(monthEl).not.toBeNull();

        expect(monthEl!.innerHTML).toEqual("April");
    });
});