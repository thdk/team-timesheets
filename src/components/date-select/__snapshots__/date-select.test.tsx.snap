// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelect should render 1`] = `
<DocumentFragment>
  <div
    class="flex horizontal"
  >
    <div
      class="first formField"
    >
      <div
        class="date-select-year mdc-select mdc-select--outlined"
        role="listbox"
      >
        <div
          class="mdc-select__anchor"
        >
          <i
            class="mdc-select__dropdown-icon"
          />
          <div
            aria-disabled="false"
            aria-haspopup="listbox"
            class="mdc-select__selected-text"
            role="button"
            tabindex="-1"
          >
             
          </div>
          <div
            class="mdc-notched-outline mdc-notched-outline--upgraded"
          >
            <div
              class="mdc-notched-outline__leading"
            />
            <div
              class="mdc-notched-outline__notch"
            >
              <span
                class="mdc-floating-label"
                for="select-Year"
                style="transition-duration: 0s;"
              >
                Year
              </span>
            </div>
            <div
              class="mdc-notched-outline__trailing"
            />
          </div>
          <select
            class="rmwc-select__native-control date-select-year"
            tabindex="0"
          >
            <option
              disabled=""
              selected=""
              value=""
            />
            <option
              value="2020"
            >
              2020
            </option>
            <option
              value="2019"
            >
              2019
            </option>
            <option
              value="2018"
            >
              2018
            </option>
            <option
              value="2017"
            >
              2017
            </option>
            <option
              value="2016"
            >
              2016
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="formField"
    >
      <div
        class="date-select-month mdc-select mdc-select--outlined"
        role="listbox"
      >
        <div
          class="mdc-select__anchor"
        >
          <i
            class="mdc-select__dropdown-icon"
          />
          <div
            aria-disabled="false"
            aria-haspopup="listbox"
            class="mdc-select__selected-text"
            role="button"
            tabindex="-1"
          >
             
          </div>
          <div
            class="mdc-notched-outline mdc-notched-outline--upgraded"
          >
            <div
              class="mdc-notched-outline__leading"
            />
            <div
              class="mdc-notched-outline__notch"
            >
              <span
                class="mdc-floating-label"
                for="select-Month"
                style="transition-duration: 0s;"
              >
                Month
              </span>
            </div>
            <div
              class="mdc-notched-outline__trailing"
            />
          </div>
          <select
            class="rmwc-select__native-control date-select-month"
            tabindex="0"
          >
            <option
              disabled=""
              selected=""
              value=""
            />
            <option
              value="0"
            >
              January
            </option>
            <option
              value="1"
            >
              February
            </option>
            <option
              value="2"
            >
              March
            </option>
            <option
              value="3"
            >
              April
            </option>
            <option
              value="4"
            >
              May
            </option>
            <option
              value="5"
            >
              June
            </option>
            <option
              value="6"
            >
              July
            </option>
            <option
              value="7"
            >
              August
            </option>
            <option
              value="8"
            >
              September
            </option>
            <option
              value="9"
            >
              October
            </option>
            <option
              value="10"
            >
              November
            </option>
            <option
              value="11"
            >
              December
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
